.github/workflows/CI.yml

```diff
@@ -119,6 +119,21 @@ jobs:
         with:
           file: lcov.info
 
+  slack:
+      name: Notify Slack Failure
+      needs: test
+      runs-on: ubuntu-latest
+      if: always() && github.event_name == 'schedule'
```

Comment:
Since we have notifications on schedules jobs (which is what we want), we also need to add a daily scheduled job to the `on:` section above

Something like

```yaml
on:
  push: 
    ...
  schedule:
    - cron: '0 2 * * *'  # Daily at 2 AM UTC
```